catroid/src/main/java/org/catrobat/catroid/ui/ProjectListActivity.java

```diff
@@ -75,4 +100,50 @@ public void handleAddButton(View view) {
 		NewProjectDialogFragment dialog = new NewProjectDialogFragment();
 		dialog.show(getSupportFragmentManager(), NewProjectDialogFragment.TAG);
 	}
+
+	private ProjectUnzipAndImportTask.ProjectUnzipAndImportListener projectUnzipAndImportListener =
+			new ProjectUnzipAndImportTask.ProjectUnzipAndImportListener() {
+				@Override
+				public void onImportFinished(boolean success) {
+					if (!success) {
+						ToastUtil.showError(ProjectListActivity.this, R.string.error_import_project);
+					}
+					loadFragment(new ProjectListFragment());
+				}
+			};
+
+	private ProjectImportTask.ProjectImportListener projectImportListener =
+			new ProjectImportTask.ProjectImportListener() {
+				@Override
+				public void onImportFinished(boolean success) {
+					if (!success) {
+						ToastUtil.showError(ProjectListActivity.this, R.string.error_import_project);
+					}
+					loadFragment(new ProjectListFragment());
+				}
+			};
+
+	private void importProjectFromExternal(String path, Intent intent) {
```

Comment:
Well, I don't think the import method in ProjectListFragment would work because it implemented to receive a Uri with scheme of "file". Whereas almost all apps share file through content URI.

Reply:
Yes, it is possible, your new code is exactly the same as in ProjectListFragment, except the "file" check and the definition of the src file.
And I think that's solvable with "Uri.fromFile(...);".
And if not, have a look at the FilePickerActivity and ProjectListFragment.

Reply:
Okay, I'll have a look at how to can solve it. Thanks!

Reply:
Is the "file" check really required?

Reply:
Which scheme does the data have? Maybe a content URI?
I will ask @84n4n4 about this check. (He wrote this code)

Reply:
i cant recall exactly why this check is there. but the whole thing was a mess to get to work properly for all android versions.

anyways:
- remove the check
- dont duplicate code, use the one in fragment that is there
- test on all supported major android versions!

Reply:
> Which scheme does the data have? Maybe a content URI?
Yup, the scheme for the implicit intents is content Uri. 
The filePickerActivity returns scheme in file Uri.

Reply:
@84n4n4 Oh. It might be an issue because I don't have access to all android versions :P (excluding emulators). 
Is it better to add an argument to the check? You must have kept it there for some reason.